tmt/steps/provision/artemis.py

```diff
@@ -347,7 +350,7 @@ def options(cls, how: Any = None) -> List[tmt.options.ClickOptionDecoratorType]:
                 help=f'A factor for exponential API retry backoff, '
                      f'{DEFAULT_RETRY_BACKOFF_FACTOR} by default.',
                 ),
-            ]) + super().options(how)
+            ]) + super().options(how) + cls._guest_class.options()
```

Comment:
* I'd definitely add `_guest_class` attribute to the base class (`tmt.steps.provision.ProvisionPlugin`) to begin with, even without a value because pretty much every derived plugin provides its own class (except `connect` which is perfectly happy with the default `GuestSsh`). So I'd start with `_guest_class: Type[Guest] = Guest` in `ProvisionPlugin`.
* `GuestSsh` is definitely not the correct class - note that all plugins except `connect` have their own subclass of `Guest`. In this case, it should definitely be `GuestArtemis` (and `GuestTestcloud` in `testcloud.py`, and so on).

To build on top of that, I'd suggest a few more extra steps:

* add `options()` method to `Guest` class, even if it'd be returning an empty list.
* add `options()` method to `GuestSsh`, to return `super().options()` plus the new `--ssh-option`
* add custom `options()` method to `ProvisionPlugin` class, to return `super().options()` plus `self._guest_class.options()`

After this, you wouldn't need to spike every plugin with extra `+ cls._guest_class.options()`, because:

* all plugins would keep calling `super().options(how)` as they do now, nothing new there.
* `PreparePlugin.options()` would be clalled (thanks to `super()`), and return its own options (probably none at this point) plus those it gets from `self._guest_class.options()` - this would be new.
* all plugins using SSH-reachable guest classes derive their guest classes from `GuestSsh` and therefore would get `--ssh-option` automatically: `ProvisionArtemis.options()` calls `ProvisionPlugin.options()` which calls `self._guest_class.options()`, which is `GuestArtemis.options()` and that's `GuestSsh.options()`. The dummy `Guest.options()` returning an empty list would make it work with guests and plugins that don't use SSH.

Reply:
Hi. Thanks for heads up and review.
Should be done.
Please take a look.